From c8f73f57d63779cfaad140e0521be40972fbec8f Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@t-online.de>
Date: Wed, 1 Sep 2021 16:44:13 +0200
Subject: [PATCH] meson: make gir buildable

yocto/oe is capable of building gobject introspection despite cross-compiling.
add an option to enable gir build even if cross-compiling

Signed-off-by: Markus Volk <f_l_k@t-online.de>
---
 libgimp/meson.build | 2 +-
 meson_options.txt   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libgimp/meson.build b/libgimp/meson.build
index 9e7b4fbbe3..99614733d7 100644
--- a/libgimp/meson.build
+++ b/libgimp/meson.build
@@ -384,7 +384,7 @@ libgimpui_introspectable_files = [
   libgimpwidgets_introspectable,
 ]
 
-if meson.is_cross_build()
+if meson.is_cross_build() and not get_option('introspection')
   have_gobject_introspection = false
   gobject_introspection_warning = '\n      '.join(['',
     'GObject Introspection is disabled while cross-compiling because',
diff --git a/meson_options.txt b/meson_options.txt
index 7b6ad78045..c29fdb5b40 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -52,7 +52,7 @@ option('headless-tests',    type: 'feature', value: 'auto', description: 'Use xv
 option('gtk-doc',           type: 'boolean', value: true,   description: 'Build developer documentation')
 option('gtk-doc-app',       type: 'boolean', value: false,  description: 'Build developer documentation for app')
 option('g-ir-doc',          type: 'boolean', value: false,  description: 'Build developer documentation for introspected API')
-
+option('introspection',   type: 'boolean', value: false,  description: 'GIR is buildable even if crosscompiling')
 option('linux-input',       type: 'feature', value: 'auto', description: 'Linux input event controller module')
 option('vec-icons',         type: 'boolean', value: true,   description: 'Use vector icons rather than raster ones')
 
-- 
2.25.1

